### ISSUE TITLE
Record constant variables not public input in generated constructor function

### ISSUE THREAD

**Issue by AnHeuermann** [OpenModelica/OpenModelica#9014]
### Description

For the record

```modelica
record R
  constant Integer i=5;
  constant Integer j;
  final constant Integer k=42;
end R;
```

the Frontend will generate the record constructor

```modelica
function R "Automatically generated record constructor for R"
  protected Integer i = 5;      // <---- illegal ?
  input Integer j;
  protected Integer k = 42;
  output R res;
end R;
```

According to the [Modelica Spec 3.5 section 12.6 Record Constructor Functions](https://specification.modelica.org/maint/3.5/functions.html)

> • Component declarations which do not allow a modification (such as final parameter Real) are declared as protected components in the record constructor function.
• Prefixes (constant, parameter, final, discrete, …) of the remaining record components are removed.
• The prefix input is added to the public components of the record constructor function.

only `k` is allowed to be protected.
`i` should be a input variable with a start value.

```modelica
  input Integer i = 5;
```

### Steps to Reproduce

Instantiate

```modelica
model mwe
  record R
    constant Integer i=5;
    constant Integer j;
    final constant Integer k=42;
  end R;

  function myFunc
    input R inR = R(j=4);
    output Integer i;
    output Integer j;
  algorithm
    i := inR.i;
    j := inR.j;
   end myFunc;
 
  Integer i,j;
  R myR(j=6);
equation
  (i,j) = myFunc(myR);
end mwe;
```

### Version and OS

 - OpenModelica Version: OpenModelica v1.20.0-dev-81-ge6238a1cc4
 - OS: Windows 11

### Additional Context

Could be part of a solution for https://github.com/OpenModelica/OpenModelica/issues/8463.

**Comment by AnHeuermann** [comment]
Strangely the flat Modelica code for the example `package Demo`  from 12.6 of the spec, which has a similar structure, is generated correctly:

```modelica
record Record2
  import Modelica.Math.*;
  extends Record1;
  final constant Real c1 = 2.0;
  constant Real c2;
  parameter Integer n1 = 5;
  parameter Integer n2;
  parameter Real r1 "comment";
  parameter Real r2 = sin(c1);
  final parameter Real r3 = cos(r2);
  Real r4;
  Real r5 = 5.0;
  Real r6[n1];
  Real r7[n2];
end Record2;
```

```modelica
function Example.Record2 "Automatically generated record constructor for Example.Record2"
  input Real r0 = 0.0;
  protected Real c1 = 2.0;
  input Real c2;
  input Integer n1 = 5;
  input Integer n2;
  input Real r1;
  input Real r2 = 0.9092974268256817;
  protected Real r3 = 0.6143002821164822;
  input Real r4;
  input Real r5 = 5.0;
  input Real[5] r6;
  input Real[n2] r7;
  output Record2 res;
end Example.Record2;
```

**Comment by AnHeuermann** [comment]
Fixed with #9016

![image](https://user-images.githubusercontent.com/38031952/170215484-02414f79-dca1-441c-8318-57dffdea523c.png)
